python/Completion.py

```diff
@@ -27,3 +40,10 @@ def get_completions(self, column, partialWord):
                 vim_completions.append(complete)
 
         return vim_completions
+    def to_dictionary_keyed_by(self, key_by, completions):
+        completion_dictionary = {}
+        if completions is not None:
+            for completion in vim.eval(completions):
+                completion_dictionary[completion.get(key_by)] = completion
+
+        return completion_dictionary
```

Comment:
Thanks, I thought you might say that. I'll have a look at implementing in Vim Script instead.